### ISSUE TITLE
Date insert in array-format bind parameters with returning value

### ISSUE THREAD

**Issue by ryandao** [oracle/node-oracledb#151]
This is the exact same issue as #121, and I believe it's not fully fixed. When I do an insert with a date parameter and use `returning`, the `NJS-012: encountered invalid bind datatype` error is thrown. It only happens for bind parameters in array format like this:

    oracledb.getConnection({
      user          : dbConfig.user,
      password      : dbConfig.password,
      connectString : dbConfig.connectString
    }, function(err, con) {
      var table = 'create table test_binds ( num number(4), str varchar2(32), dt date )';
      con.execute(table, function(err) {
        if (err) { return console.log(err.message); }
        
        con.execute(insert, params, function(err) {
          if (err) { return console.log(err.message); }
          var insert = 'insert into test_binds (num, str, dt) values (:0, :1, :2) returning num into :3';
          var params = [
            123,
            'str',
            new Date(),
            outNum: { type: oracledb.NUMBER, dir: oracledb.BIND_OUT }
          ];
          con.execute(insert, params, function(err) {
            if (err) { return console.log(err.message); }
            console.log('insert ok 0');
          });
        });
      });
    });

If I change `params` to `{ 0: 123, 1: 'str', 2: new Date(), 3: outNum: { type: oracledb.NUMBER, dir: oracledb.BIND_OUT } }` or remove the date value, it works.

**Comment by cjbj** [comment]
:(

**Comment by cjbj** [comment]
Just a head's up that mixing array and named bind styles will throw an error in node-oracledb 1.0.  You will need to delete the text ' outNum: ' and use:
```
[ 123, 'str', new Date(), { type: oracledb.NUMBER, dir: oracledb.BIND_OUT } ]
```

(The error message thrown in 1.0 could be improved, but that is pending)

**Comment by cjbj** [comment]
@ryandao this is fixed in node-oracledb 1.0.  Try it again. Thanks for the report.  Also see my previous syntax caveat.
